Sources/MapboxCoreNavigation/Router.swift

```diff
@@ -219,8 +234,8 @@ extension InternalRouter where Self: Router {
             guard case let .success(response) = result, let self = self else {
                 return
             }
-            
-            self.routeProgress.refreshRoute(with: response.route, at: location)
+            self.indexedRouteResponse = .init(routeResponse: response, routeIndex: self.indexedRouteResponse.routeIndex)
+            self.routeProgress.refreshRoute(with: self.indexedRouteResponse.currentRoute!, at: location)
```

Comment:
nit: I think precondition with a clear message would be better here in case something goes wrong.